app/src/main/kotlin/no/nav/k9punsj/innsending/RestInnsendingClient.kt

```diff
@@ -0,0 +1,200 @@
+
+        val søknad = søknadJson.somPunsjetSøknad(
+            versjon = punsjetSøknadJson["versjon"].asText(),
+            saksbehandler = punsjetSøknadJson["saksbehandler"].saksbehandler(),
+            brevkode = brevkode,
+            saksnummer = k9Saksnummer
+        )
+
+        // Hent k9saksnummer
+        if (k9Saksnummer.isNullOrEmpty()) {
+            val k9SaksnummerGrunnlag = HentK9SaksnummerGrunnlag(
+                søknadstype = fagsakYtelseType,
+                søker = søknad.søker.toString(),
+                pleietrengende = søknad.pleietrengende.toString(),
+                annenPart = søknad.annenPart.toString(),
+                journalpostId = søknad.journalpostIder.first().toString()
+            )
+            k9SakService.hentEllerOpprettSaksnummer(k9SaksnummerGrunnlag).first?.let {
+                k9Saksnummer = objectMapper().readValue<SaksnummerDto>(it).saksnummer
+
+            }
+        }
+
+        requireNotNull(k9Saksnummer) { "K9Saksnummer er null" }
+
+        // Ferdigstill journalposter
+        val søkerNavn = pdlService.hentPersonopplysninger(setOf(søknad.søker.toString()))
+        require(søkerNavn.isNotEmpty()) { throw IllegalStateException("Fant ikke søker i PDL") }
+        val bruker = FerdigstillJournalpost.Bruker(
+            identitetsnummer = søknad.søker,
+            navn = søkerNavn.first().navn(),
+            sak = Fagsystem.K9SAK to Saksnummer(k9Saksnummer)
+        )
+
+        val ferdigstillJournalposter = søknad.journalpostIder.map { journalpostId ->
+            safGateway.hentFerdigstillJournalpost(journalpostId = journalpostId)
+        }.filterNot { ferdigstillJournalpost ->
+            ferdigstillJournalpost.erFerdigstilt.also {
+                if (it) {
+                    logger.info("JournalpostId=[${ferdigstillJournalpost.journalpostId}] er allerede ferdigstilt.")
+                }
+            }
+        }.map { it.copy(
+            bruker = bruker,
+            sak = FerdigstillJournalpost.Sak(
+                sakstype = "FAGSAK",
+                fagsaksystem = "K9",
+                fagsakId = k9Saksnummer
+            )
+        ) }
+
+        val manglerAvsendernavn = ferdigstillJournalposter.filter { it.manglerAvsendernavn() }
+
+        require(manglerAvsendernavn.isEmpty()) {
+            "Mangler avsendernavn på journalposter=[${manglerAvsendernavn.map { it.journalpostId }}]"
+        }
+
+        // Alle journalposter klare til oppdatering & ferdigstilling
+        check(ferdigstillJournalposter.all { it.kanFerdigstilles }).also {
+            logger.info("Journalposter klare for ferdigstilling: ${ferdigstillJournalposter.map { it.journalpostId }}")
+        }
+
+        ferdigstillJournalposter.forEach { ferdigstillJournalpost ->
+            dokarkivGateway.oppdaterJournalpostForFerdigstilling(correlationId, ferdigstillJournalpost)
+            dokarkivGateway.ferdigstillJournalpost(ferdigstillJournalpost.journalpostId.toString(), "9999")
```

Comment:
Fint om du kan definere en beskrivende konstant eller variabel for denne verdien.

Reply:
Denne klassen skall skrives om etter prodsetting, vi må bara få den på plass sedan prioritere å skrive tester på hele denne flowen.... SEN kan vi refaktorere :)